docs: add note that only system job preemption is available in OSS (#7936)
hashicorp/nomad: website/pages/docs/internals/scheduling/preemption.mdx

# Details

> ~> **Enterprise Functionality** System job preemption is available as an OSS
> ~> feature, while Batch and Service job preemption is only available in the
> ~> Enterprise offering.

Reviewer: ```suggestion
~> **Enterprise Functionality** System job preemption is available as an Open Source
 feature, while Batch and Service job preemption are only available as 
 Enterprise features.
```
